source/_posts/2019-03-11-upgrading-to-embark-4.md

```diff
@@ -0,0 +1,73 @@
+
+Historically, Embark has come with a special Webpack pipeline because it made your work, as a user, easier. Indeed, it enabled you to use simple "magic" imports, such as `import SimpleStorage from "Embark/contracts/SimpleStorage";`. It also did the connection to Web3 for you.
+
+However, we discovered the hard way that those advantages ended not being worth the price of performance that you pay, when you can't use an optimized pipeline, like create-react-app or Angular CLI.
+
+Thereby, we announce that you can now use the file pipeline that YOU want. Also, you can STILL use Embark's pipeline if you are fine with what is currently there.
+
+So, as you just read, it's not really a breaking change, well  that is, unless you decide you want to convert you Dapp from Embark's pipeline to the one of your choice.
+
+If you don't plan on starting a project with another pipeline or don't plan of converting, you can skip to [the next section](#New-Web3-plugin).
+
+### Converting to another pipeline
+
+The first thing you might want to do is change or add the new `generationDir` property in `embark.json`. This property tells Embark where to generation the various artifacts that you will need to build your Dapp. Most of those artifacts were already generated before, but were inside the `.embark/` folder. 
+
+Since most file pipelines need those files inside the source folder, we're going to put `generationDir` as your selected pipeline's source folder. For example,  create-react-app (CRA) has `src/` as source folder.
+
+Afterwards, we need to convert all "magic" imports to relative imports.
+
+The first one is the EmbarkJS import. The "magic" import is `"Embark/EmbarkJS"`. We need to convert that to the relative path. The `embarkjs.js` script is at the root of  `embarkArtifacts/`, so you need to convert it to something like `"./embarkArtifacts/embarkjs"`.
+
+Second ones are the contracts. They used to look like `"Embark/contract/ContractName"`. They are now located inside `embarkArtifactscontracts/`.
+
+Third one, there used to be `"Embark/web3"`, but we removed it. Don't worry, we are not removing web3 support, far from it. We actually just got rid of that useless import that didn't do much. All you have to do is use the global `web3` object that is avilable everywhere in the Dapp.
+
+Then, you can move your Embark files and configs inside a project created by the pipeline of your choice.
+
+### New project with another pipeline
+
+If you are starting your project from scratch, you have two options.
+
+The easier one is to use our new Embark CRA template (TODO add template URl). It sets up a simple Embark project for you with the all of the files you are used to. The only slight difference is that the config files are in `embarkConfig` to make sure they don't clash with CRA's config. It's already set up for you to start and use. You only need to do `embark run` and in another terminal `yarn start` or `npm run start` to start CRA.
```

Comment:
```suggestion
The easiest option is to use our new Embark CRA template (TODO add template URl). It sets up a simple Embark project with all of the familiar files present in an Embark 3.x Dapp, with one minor difference: the config files are located in an `embarkConfig/` folder in the root of the Dapp to make sure they don't clash with CRA's config folder/files.
```
To get started with Embark's CRA template,
```
npx embark-react my-app
cd my-app
yarn start // or alternatively, npm run start
```
Then, in another terminal,
```
cd my-app
embark run
```
That's it!